sql-plugin/src/main/scala/com/nvidia/spark/rapids/Plugin.scala

```diff
@@ -531,7 +531,7 @@ class RapidsExecutorPlugin extends ExecutorPlugin with Logging {
       // Checks if the current GPU architecture is supported by the
       // spark-rapids-jni and cuDF libraries.
       // Note: We allow this check to be skipped for off-chance cases.
-      if (!conf.skipGpuArchCheck) {
+      if (!conf.skipGpuArchCheck && conf.isSqlEnabled && conf.isSqlExecuteOnGPU) {
```

Comment:
Since `sqlEnabled` is not a `startupOnly` conf I think we should not make it a prerequisite for calling `validateGpuArchitecture`

```suggestion
      if (!conf.skipGpuArchCheck && conf.isSqlExecuteOnGPU) {
```